File: react-front-end-app/src/services/scaleService.js
import apiClient from "@/config/axiosConfig";

// handles fetching scale data from the API

export function getAllScales() {
    return apiClient.get('/scales');
}

export function getAllScalesByDifficulty(difficulty) {
    return apiClient.get(`/scales/difficulty/${difficulty}`);
}

export function getScaleById(id) {
    return apiClient.get(`/scales/${id}`);
}

export function getTotalScales() {
    return apiClient.get('/scales/count');
}

File: java-spring-boot-back-end-app/src/main/java/com/github/joshliford/amplifyguitar/service/ScaleService.java
package com.github.joshliford.amplifyguitar.service;

import com.github.joshliford.amplifyguitar.exception.ResourceNotFoundException;
import com.github.joshliford.amplifyguitar.model.Difficulty;
import com.github.joshliford.amplifyguitar.model.Scale;
import com.github.joshliford.amplifyguitar.repository.ScaleRepository;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class ScaleService {

    private final ScaleRepository scaleRepository;

    public ScaleService(ScaleRepository scaleRepository) {
        this.scaleRepository = scaleRepository;
    }

    public List<Scale> getAllScales() {
        return scaleRepository.findAllByOrderByTitleAsc();
    }

    public List<Scale> getAllScalesByDifficulty(Difficulty difficulty) {
        return scaleRepository.findByDifficultyOrderByTitleAsc(difficulty);
    }

    public Scale getScaleById(Integer id) {
        return scaleRepository.findById(id)
                .orElseThrow(() -> new ResourceNotFoundException("Scale not found with id: " + id));
    }

    public Long getTotalScales() {
        return scaleRepository.count();
    }

}
